extensions/arc/deployment/src/main/java/io/quarkus/arc/deployment/AdditionalBeanBuildItem.java

```diff
@@ -61,9 +61,6 @@ public AdditionalBeanBuildItem(Class<?>... beanClasses) {
     }
 
     AdditionalBeanBuildItem(List<String> beanClasses, boolean removable, DotName defaultScope) {
-        if (beanClasses.isEmpty()) {
-            throw new IllegalArgumentException("No bean classes specified");
-        }
```

Comment:
Wouldn't the right fix be to not create the build item in the first place? I think it was the initial design.

Reply:
That is what I thought initially as well, but then I thought that the consumer of `AdditionalBeanBuildItem` doesn't really care - for them it's much easier to just pass whatever collection they have and get on with business.
Either way we want to do it, I'm fine with it - it's just that I think this way would avoid similar surprises in the future

Reply:
That's fine with me. It's easier for `AdditionalBeanBuildItem` producers.